File: 0974-subarray-sums-divisible-by-k/0974-subarray-sums-divisible-by-k.py
class Solution:
    def subarraysDivByK(self, nums: List[int], k: int) -> int:
        ans = tot = 0
        remain = defaultdict(int)
        remain[0] = 1
        for num in nums:
            tot += num
            mod = tot % k
            ans += remain[mod]
            remain[mod] += 1
        
        return ans

File: 0974-subarray-sums-divisible-by-k/0974-subarray-sums-divisible-by-k.go
func subarraysDivByK(nums []int, k int) int {
    var modOccur = make(map[int]int)
    var currSum, counts = 0, 0

    modOccur[0] = 1
    for _, num := range nums {
        currSum = (currSum + num % k + k) % k
        // var mod = int(math.Abs(float64()))
        fmt.Println(currSum)
        counts += modOccur[currSum]
        modOccur[currSum]++
    }
    return counts
}